source/Halibut.Tests/Support/LatestClientAndLatestServiceBuilder.cs

```diff
@@ -356,301 +277,82 @@ async Task<IClientAndService> IClientAndServiceBuilder.Build(CancellationToken c
-                    disposableCollection.Add(webSocketSslCertificate);
-
-                    portForwarder = portForwarderFactory?.Invoke(webSocketListeningInfo.WebSocketListeningPort);
-
-                    clientUri = new Uri($"wss://localhost:{portForwarder?.ListeningPort ?? webSocketListeningInfo.WebSocketListeningPort}/{webSocketListeningInfo.WebSocketPath}");
-                    clientUrsToPoll.Add(clientUri);
-                }
-
-                if (service != null)
-                {
-                    foreach (var clientUriToPoll in clientUrsToPoll)
-                    {
-                        service.Poll(
-                            serviceUri,
-                            new ServiceEndPoint(clientUriToPoll, serviceTrustsThumbprint, httpProxyDetails, service.TimeoutsAndLimits),
-                            CancellationToken.None);
-                    }
-                }
-            }
-            else if (ServiceConnectionType == ServiceConnectionType.Listening)
-            {
-                int listenPort;
-                if (service != null)
-                {
-                    service.Trust(serviceTrustsThumbprint);
-                    listenPort = service.Listen();
-                }
-                else
-                {
-                    var dummyTentacle = new TCPListenerWhichKillsNewConnections();
-                    disposableCollection.Add(dummyTentacle);
-                    listenPort = dummyTentacle.Port;
-                }
-
-                portForwarder = portForwarderFactory?.Invoke(listenPort);
-                if (portForwarder != null)
-                {
-                    listenPort = portForwarder.ListeningPort;
-                }
-
-                serviceUri = new Uri("https://localhost:" + listenPort);
-            }
-            else
-            {
-                throw new NotSupportedException();
-            }
-
-            if (portForwarderReference != null && portForwarder != null)
-            {
-                portForwarderReference.Value = portForwarder;
-            }
-
-            return new ClientAndService(client, clientUri, service, serviceUri, clientTrustsThumbprint, portForwarder, disposableCollection, httpProxy, httpProxyDetails, cancellationTokenSource);
-        }
-
-        IPendingRequestQueueFactory CreatePendingRequestQueueFactory(ILogFactory octopusLogFactory)
-        {
-            if (pendingRequestQueueFactory != null)
-            {
-                return pendingRequestQueueFactory(octopusLogFactory);
+                clientBuilder.WithProxyDetails(httpProxyDetails);
+                serviceBuilder.WithProxyDetails(httpProxyDetails);
             }
 
-            var pendingRequestQueueFactoryBuilder = new PendingRequestQueueFactoryBuilder(octopusLogFactory);
-
-            if (this.pendingRequestQueueFactoryBuilder != null)
+            var client = await clientBuilder.Build(cancellationToken);
+            if (client.ListeningUri is not null)
```

Comment:
If we are testing "Polling", then the client is listening, and the service needs to know which endpoint to "Poll"